articles/understanding-differential-privacy/index.md

```diff
@@ -0,0 +1,280 @@
+
+### Differential Privacy
+
+#### Former definition
+
+*Anything that can be learned from a statistical database can be learned without accessing the database*
+
+This definition does not hold always. We need to consider two things, first is people and the second is data. We need to protect people's rights as well as predict rightly using their data. In certain cases, the whole database must be kept private, to protect the information, because there are chances of data leaks.
+
+#### Modern definition
+
+*It’s a promise made by a data holder to a data subject, that “You will not be affected, adversely or otherwise, by allowing your data to be used for study analysis, no matter what other studies, datasets or information is available”
+-Cynthia Dwork (Godfather of Differential Privacy)*
+
+A simple way to explain the modern definition is
+
+Assume, we have a small database of 5000 entries with 0 or 1 as a value for each row, specifying certain property like 'people with cancer' as 1, and 'people without cancer' as 0. So, here our goal is "Even if we remove detail of 1 person, the query of the database must not change” then the privacy of the information is protected. We will see its implementation very shortly.
+
+![Differential Privacy working](/engineering-education/understanding-differential-privacy/differential_privacy.png)
+
+[Image source](https://www.winton.com/research/using-differential-privacy-to-protect-personal-data)
+
+### Differencing attacks
+
+In continuation with the previous example, let's say we want to find if a specific person has cancer or not, so what we do is, find the sum of all the values, and then find the sum of all values after removing that person. On finding the difference between both the summations, we get the exact value of that person, thus we will know if he has cancer or not. This is one of the simplest differencing attacks, using summation.
+
+Similarly, there are other types of differencing attacks using various other functions like mean(), median(), and so on. This measure of how much data is leaked through a query can be measured with **sensitivity.**
+
+### Implementing differential privacy
+
+Now, we are going to implement Differential privacy for a simple database query. The database has only 1 column with boolean types. The boolean type denotes if a person possesses some private attribute or not (example, if a person has a particular disease or not). And, we are going to learn, if the database is differentially private or not.
+
+#### Creating a database
+
+Create a simple database containing 5000 entries consisting of values True/False (1/0). Here, we randomly generate boolean values using `torch.rand()`
+
+```python
+  import torch
```

Comment:
A beginner might have not used PyTorch before. So it would be helpful to add a couple of lines explaining how PyTorch can be installed